rust/private/rustc.bzl

```diff
@@ -2105,7 +2105,7 @@ def _add_native_link_flags(args, dep_info, linkstamp_outs, ambiguous_libs, crate
     if toolchain.target_os == "windows":
         make_link_flags = _make_link_flags_windows_msvc if toolchain.target_triple.abi == "msvc" else _make_link_flags_windows_gnu
         get_lib_name = get_lib_name_for_windows
-    elif toolchain.target_os.startswith(("mac", "darwin", "ios")):
+    elif toolchain.target_os.startswith(("mac", "ios")):
```

Comment:
I was surprised that `mac` was already being handled. Is that to handle custom values passed in via `target_json`?

Reply:
I don't think you shouldn't delete `darwin`, you should just be adding "macos" as an option. And yeah, we should continue to support the case where folks using `target_json` explicitly wrote "darwin".

Reply:
Addressed in the latest commit

Sorry, I force pushed so there's not a commit to see the latest changes – it was easier to `git reset` and undo the changes.